pkg/monitor/cluster/podconditions.go

```diff
@@ -59,6 +62,14 @@ func (mon *Monitor) _emitPodConditions(ps *corev1.PodList) {
 			continue
 		}
 
+		if p.Status.Reason == Preempting {
+			for _, cs := range p.Status.ContainerStatuses {
+				if cs.State.Terminated != nil && cs.State.Terminated.Reason == Completed {
+					continue
+				}
+			}
```

Comment:
I think this is more a question about the acceptance criteria of the WI rather than the code, but should we just ignore any Pod marked as `Preempting`? We know that, eventually, this Pod will have all it's container Terminated - so do we actually need to check container status (in this case, that any 1 container in this Pod has been terminated) before ignoring, or can we ignore that race condition all-together?

Reply:
Agree with the suggestion, and will make changes to skip all pods which have reason Preempting. Thank you!